In [3]:
import pandas as pd
import nltk
from nltk.tokenize import word_tokenize
from nltk import pos_tag
import matplotlib.pyplot as plt
from tqdm import tqdm
import swifter

#required resources; download once
#nltk.download('punkt')
#nltk.download('averaged_perceptron_tagger')

/opt/anaconda3/envs/studie/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
df = pd.read_csv('data/995,000_row_cleaned.csv')

/var/folders/5h/8kz4p9fx43v2v3g991x788nw0000gn/T/ipykernel_19508/1234656882.py:1: DtypeWarning: Columns (0,1) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('data/995,000_row_cleaned.csv')


In [ ]:
# Totalt antal ord i content
total_words = df["content"].swifter.progress_bar(True).apply(lambda x: len(str(x).split())).sum()
print("Total words in dataset:", total_words)

# Funktion til at finde alle unikke ord i en text
def extract_unique_words(text):
    return set(str(text).split())

# Apply function efficiently using swifter
unique_word_sets = df["content"].swifter.progress_bar(True).apply(extract_unique_words)

# Combine all sets and count unique words
total_unique_words = len(set().union(*tqdm(unique_word_sets)))

print("Total unique words in dataset:", total_unique_words)

# Procentdel af unikke ord pr. ord
print("Unique words per word:", (total_unique_words / total_words) * 100)

Pandas Apply: 100%|██████████| 995000/995000 [01:05<00:00, 15143.61it/s]


Total words in dataset: 263245932


100%|█████████▉| 994839/995000 [00:05<00:00, 53672.22it/s] 

Pandas Apply: 100%|██████████| 995000/995000 [00:37<00:00, 26413.46it/s]
Total words in dataset: 263245932
Pandas Apply: 100%|██████████| 995000/995000 [06:29<00:00, 2556.07it/s] 
Total unique words in dataset: 1726502
Unique words per word: 0.6558513504398616